From: hololeap <hololeap@protonmail.com>
Signed-off-by: hololeap <hololeap@protonmail.com>

This enables parallelism on the executables and turns it on by default.

diff -urN ghcide-1.5.0.1/ghcide.cabal ghcide-1.5.0.1-r1/ghcide.cabal
--- ghcide-1.5.0.1/ghcide.cabal	2021-12-03 01:15:49.393629703 -0700
+++ ghcide-1.5.0.1-r1/ghcide.cabal	2021-12-03 01:29:42.373640400 -0700
@@ -297,7 +297,7 @@
                 -rtsopts
                 -- disable idle GC
                 -- increase nursery size
-                "-with-rtsopts=-I0 -A128M"
+                "-with-rtsopts=-I0 -A128M -N"
     main-is: Main.hs
     build-depends:
         hiedb,
@@ -404,7 +404,17 @@
           record-dot-preprocessor,
           record-hasfield
     hs-source-dirs: test/cabal test/exe test/src bench/lib
-    ghc-options: -threaded -Wall -Wno-name-shadowing -O0 -Wno-unticked-promoted-constructors
+    ghc-options:
+                -threaded
+                -Wall
+                -Wno-name-shadowing
+                -O2
+                -Wno-unticked-promoted-constructors
+                -- allow user RTS overrides
+                -rtsopts
+                -- disable idle GC
+                -- increase nursery size
+                "-with-rtsopts=-I0 -A128M -N"
     main-is: Main.hs
     other-modules:
         Development.IDE.Test
@@ -458,7 +468,16 @@
         tasty-hunit,
         text
     hs-source-dirs: bench/lib bench/exe test/src
-    ghc-options: -threaded -Wall -Wno-name-shadowing -rtsopts
+    ghc-options:
+                -threaded
+                -Wall
+                -Wno-name-shadowing
+                -O2
+                -- allow user RTS overrides
+                -rtsopts
+                -- disable idle GC
+                -- increase nursery size
+                "-with-rtsopts=-I0 -A128M -N"
     main-is: Main.hs
     other-modules:
         Development.IDE.Test
